Buildings/Controls/OBC/Utilities/PIDWithAutotuning/Relay/HalfPeriodRatio.mo

```diff
@@ -0,0 +1,212 @@
+          20},{60,-6},{18,-6},{18,-88},{28,-88}}, color={0,0,127}));
+  connect(gretmaxtOntOff.u1, AddtOntOff.y) annotation (Line(points={{-22,0},{
+          -40,0},{-40,-30},{-58,-30}}, color={0,0,127}));
+  connect(gretmaxtOntOff.u2, minLen.y) annotation (Line(points={{-22,8},{-40,8},
+          {-40,10},{-58,10}},color={0,0,127}));
+  connect(gretOntOff.u1, tmin.y) annotation (Line(points={{-22,40},{-58,40}},
+                                       color={0,0,127}));
+  connect(mul.u1, tmin.y) annotation (Line(points={{-36,-24},{-44,-24},{-44,40},
+          {-58,40}},         color={0,0,127}));
+  connect(mul.u2, AddtOntOff.y) annotation (Line(points={{-36,-36},{-40,-36},{
+          -40,-30},{-58,-30}}, color={0,0,127}));
+  connect(mul.y, tIncrease.u1)
+    annotation (Line(points={{-12,-30},{28,-30}}, color={0,0,127}));
+  connect(tDecrease.u1, mul.y) annotation (Line(points={{28,-80},{10,-80},{10,
+          -30},{-12,-30}}, color={0,0,127}));
+  connect(samAddtOntOff.u, mul.y) annotation (Line(points={{18,20},{10,20},{10,
+          -30},{-12,-30}},  color={0,0,127}));
+  connect(tOnSample.trigger, tChanges.y) annotation (Line(points={{-70,68},{-70,
+          60},{94,60},{94,-30},{92,-30}}, color={255,0,255}));
+  connect(triEnd, tChanges.y) annotation (Line(points={{120,-30},{92,-30}},
+                          color={255,0,255}));
+  connect(tOnSample.y, mintOntOff.u1) annotation (Line(points={{-58,80},{-52,80},
+          {-52,-64},{-22,-64}}, color={0,0,127}));
+  connect(tChanges.y, tOffSample.trigger) annotation (Line(points={{92,-30},{94,
+          -30},{94,-54},{-70,-54},{-70,-58}}, color={255,0,255}));
+  connect(tDecrease.y, tChanges.u2) annotation (Line(points={{52,-80},{60,-80},
+          {60,-38},{68,-38}}, color={255,0,255}));
+  annotation (defaultComponentName = "halPerRat",
+        Diagram(
+           coordinateSystem(
+           extent={{-100,-100},{100,100}})),
+        Icon(coordinateSystem(preserveAspectRatio=false, extent={{-100,-100},{80,
+            100}}),                                      graphics={
+        Rectangle(
+          extent={{-100,-100},{100,100}},
+          lineColor={0,0,127},
+          fillColor={255,255,255},
+          fillPattern=FillPattern.Solid),
+        Text(
+          extent={{-154,148},{146,108}},
+          textString="%name",
+          textColor={0,0,255})}),
+    Documentation(revisions="<html>
+<ul>
+<li>
+June 1, 2022, by Sen Huang:<br/>
+First implementation<br/>
+</li>
+</ul>
+</html>", info="<html>
+<p>This block calculates the half-period ratio of the output from a relay controller.</p>
+<h4>Main equations</h4>
+<p align=\"center\" style=\"font-style:italic;\">
+&rho; = max(t<sub>on</sub>,t<sub>off</sub>)/ min(t<sub>on</sub>,t<sub>off</sub>),
+</p>
+<p>
+where <code>t<sub>on</sub></code> and <code>t<sub>off</sub></code> are the
+length of the On period and the Off period, respectively.
+</p>
+<p>During an On period, the relay switch signal becomes True;
```

Comment:
This block does not have the relay switch input. Should this section be removed?

Reply:
Sorry for the confusion. The relay switch is referred to define the On period and the Off period. I modified the doc to avoid confusion.

Reply:
see #3291